monix-execution/js/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -0,0 +1,104 @@
+package monix.execution.misc
+
+object Local {
+
+  type Context = scala.collection.immutable.Map[Key, Option[LocalContext]]
+
+  trait LocalContext
+
+  class Key
+
+  private[this] def context(k: Key, v: Option[LocalContext]): Context =
+    scala.collection.immutable.Map(k -> v)
+
+  private[this] val localContext = ThreadLocal[Context]()
```

Comment:
N.B. I would go for the shared file, because boxing this in JS's `ThreadLocal` is not that problematic, unless you have other reasons for keeping it separate.